### PR TITLE
Run works locally. Problem in ports UI: integer types don't show corr…

### PR DESCRIPTION
**Pull request by elisabettai**
This creates the k-wave service.

related to: https://github.com/ITISFoundation/osparc-issues/issues/694

### CODE REVIEW

File: .osparc/metadata.yml [hunk 1]
```diff
@@ -13,19 +13,173 @@ authors:
...
+      input.h5: input_1
+  input_2:
+    displayOrder: 2
+    label: Compression level (-c)
+    description: Compression level
+    type: ref_contentSchema
+    contentSchema: 
+      title: Compression level
+      type: integer
+      default: 1
```
Comment (elisabettai) [inline .osparc/metadata.yml]: @pcrespov, in the UI, I don't see the default value I have here. Would you see why that happens? How can I debug this? I have it running in my local deploy.

Comment (pcrespov) [reply .osparc/metadata.yml]: at first sight it looks correct to me https://json-schema.org/understanding-json-schema/reference/numeric.html

Try this: make `input_2.label` the same as `input_2.contentSchema.title`. Perpahs there is bug ...

File: .osparc/docker-compose.overwrite.yml [hunk 2]
```diff
@@ -4,3 +4,6 @@ services:
     build:
       dockerfile: docker/custom/Dockerfile
       target: production
+      args: 
+        - KWAVE_LOGIN
```
Comment (pcrespov) [inline .osparc/docker-compose.overwrite.yml:8]: how are you passing these?? They should go as environs on the service, no? Perhaps you should use `oenvs` for that... It is under development in https://github.com/ITISFoundation/osparc-simcore/issues/3565

Comment (elisabettai) [reply .osparc/docker-compose.overwrite.yml:8]: Nope @pcrespov, I don't think this are environs of the service.
I use them during building to download the executable of this tool from a website that requires login. See here:
https://github.com/ITISFoundation/k-wave/blob/first_version/docker/custom/Dockerfile#L49

File: .osparc/metadata.yml (hunk 1 above)
Comment (elisabettai) [reply .osparc/metadata.yml]: So it didn't work. I got the impression that we don't do anything with `default` if provided within the `contentSchema`. 

And if understood this well, default is just an "annotation": https://stackoverflow.com/a/72048005

I solved by changing to:
```
  input_2:
    displayOrder: 2
    label: Compression level
    description: Compression level (-c)
    defaultValue: 0
    type: ref_contentSchema
    contentSchema: 
      title: Compression level
      type: integer
      minimum: 0
      maximum: 9
```

**Review by pcrespov — COMMENTED** [review]
👍

File: service.cli/execute.sh [hunk 3]
```diff
@@ -13,21 +13,88 @@ echo "..."
...
-# put the code to execute the service here
-# For example:
-env
-ls -al "${INPUT_FOLDER}"
+# From isolve: checking for resources limit env vars injected by osparc
+SIMCORE_NANO_CPUS_LIMIT="${SIMCORE_NANO_CPUS_LIMIT:-0}"
+echo "Setting/getting CPU limit gives this: ${SIMCORE_NANO_CPUS_LIMIT}"
+if [ "${SIMCORE_NANO_CPUS_LIMIT}" -ne "0" ]
+then
+    echo "Found NANO_CPU limits: ${SIMCORE_NANO_CPUS_LIMIT}"
```
Comment (elisabettai) [inline service.cli/execute.sh:22]: @mguidon, mainly for curiosity:
Before I had:
`echo "$INFO" "Found NANO_CPU limits: ${SIMCORE_NANO_CPUS_LIMIT}"`
But I had an `unbound variable` error in my local oSPARC deploy.

I had to remove "$INFO" to make it work. I am not sure if this is due to something that has changed with the log levels.

File: .osparc/metadata.yml (hunk 1 above)
Comment (sanderegg) [reply .osparc/metadata.yml]: well I guess this follows the node json schema @pcrespov .

**Review by sanderegg — APPROVED** [review]

**Pull request merged**